src/main/groovy/com/wooga/gradle/io/ProcessExecutor.groovy

```diff
@@ -0,0 +1,229 @@
+package com.wooga.gradle.io
+
+import com.wooga.gradle.ArgumentsSpec
+import groovy.transform.stc.ClosureParams
+import groovy.transform.stc.FromString
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.internal.ProcessOperations
+import org.gradle.process.ExecResult
+
+import java.util.function.Function
+
+/**
+ * Executes a process after having configured it through exposed methods
+ */
+class ProcessExecutor {
+
+    //------------------------------------------------------------------------/
+    // Fields
+    //------------------------------------------------------------------------/
+    // Required
+    Function<Action<? super org.gradle.process.ExecSpec>, ExecResult> function
+    String executablePath
+    // Optional
+    List<String> arguments
+    Map<String, ?> environment
+    Boolean appendToEnvironment
+    Closure closure
+    Boolean _ignoreExitValue = false
+    String workingDirectory
+    OutputStream _standardOutput, _standardError
+
+    //------------------------------------------------------------------------/
+    // Constructors
+    //------------------------------------------------------------------------/
+    private ProcessExecutor(ProcessOperations processOperations) {
+        function = { a -> processOperations.exec(a as Action<? super org.gradle.process.ExecSpec>) }
+    }
+
+    private ProcessExecutor(Project project) {
+        function = { a -> project.exec(a as Action<? super org.gradle.process.ExecSpec>) }
+    }
+
+    /**
+     * Constructs the executor from an object implementing the {@code ProcessOperations} interface
+     */
```

Comment:
Sorry again. But I can see that it is a constructor and what type it has.

Reply:
Very well.